kernel/src/kcb.rs

```diff
@@ -383,10 +396,42 @@ impl<A: ArchSpecificKcb> Kcb<A> {
         }
     }
 
+    pub fn set_pmem_affinity(&mut self, node: atopology::NodeId) -> Result<(), KError> {
+        if node == self.pmem_memory.affinity {
+            // Allocation affinity is already set to correct NUMA node
+            return Ok(());
+        }
+
+        if node < self.pmem_arenas.len() && node < atopology::MACHINE_TOPOLOGY.num_nodes() {
+            let gmanager = self
+                .pmem_memory
+                .gmanager
+                .ok_or(KError::GlobalMemoryNotSet)?;
+
+            if self.pmem_arenas[node].is_none() {
+                self.pmem_arenas[node] = Some(PhysicalMemoryArena::new(node, gmanager));
+            }
+            debug_assert!(self.pmem_arenas[node].is_some());
+            let mut arena = self.pmem_arenas[node].take().unwrap();
+            debug_assert_eq!(arena.affinity, node);
+
+            core::mem::swap(&mut arena, &mut self.pmem_memory);
+            self.pmem_arenas[arena.affinity as usize].replace(arena);
+
+            Ok(())
+        } else {
+            Err(KError::InvalidAffinityId)
+        }
+    }
+
     pub fn set_physical_memory_manager(&mut self, pmanager: TCache) {
```

Comment:
maybe rename to set_mem_manager for consistency

Reply:
Updated.